src/core/buffer/append_data.ts

```diff
@@ -18,53 +18,18 @@
  * This file allows any Buffer to push data to a QueuedSourceBuffer.
  */
 
-import {
-  Observable,
-  of as observableOf,
-} from "rxjs";
+import { Observable } from "rxjs";
 import {
   catchError,
   mergeMapTo,
 } from "rxjs/operators";
 import { MediaError } from "../../errors";
-import { ISegment } from "../../manifest";
-import { QueuedSourceBuffer } from "../source_buffers";
+import {
+  IAppendBufferInfos,
+  QueuedSourceBuffer,
+} from "../source_buffers";
 import forceGarbageCollection from "./force_garbage_collection";
 
-export interface IAppendedSegmentInfos<T> {
-  segment : ISegment;
-  initSegmentData : T|null;
-  segmentData : T;
-  segmentOffset : number;
-}
-
-/**
- * Append buffer to the queuedSourceBuffer.
- * If it leads to a QuotaExceededError, try to run our custom range
- * _garbage collector_.
- *
- * @param {Object} queuedSourceBuffer
- * @param {Object} appendedSegmentInfos
- * @returns {Observable}
- */
-function appendDataToSourceBuffer<T>(
-  queuedSourceBuffer : QueuedSourceBuffer<T>,
-  { segment, initSegmentData, segmentData, segmentOffset } : IAppendedSegmentInfos<T>
-) : Observable<void> {
-
-  let append$ : Observable<void>;
-  if (segment.isInit) {
-    append$ = initSegmentData == null ?
-      observableOf(undefined) :
-      queuedSourceBuffer.appendBuffer(initSegmentData, null, segmentOffset);
-  } else {
-    append$ = segmentData == null ?
-      observableOf(undefined) :
-      queuedSourceBuffer.appendBuffer(initSegmentData, segmentData, segmentOffset);
-  }
-  return append$;
-}
```

Comment:
That function did not serve any purpose but to play with appendBuffer's arguments.

Now that ``appendDataToSourceBufferWithRetries`` aks directly for the right infos, this is not needed.